src/Adapter/Backup/BackupProvider.php

```
@@ -0,0 +1,64 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Backup;
+
+use PrestaShop\PrestaShop\Adapter\Entity\PrestaShopBackup;
+use PrestaShop\PrestaShop\Core\Backup\BackupProviderInterface;
+
+/**
+ * Class BackupProvider is responsible for providing available backups
+ *
+ * @internal
+ */
+final class BackupProvider implements BackupProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getBackups()
+    {
+        $backupPath = @opendir(PrestaShopBackup::getBackupPath());
```

don't "@" it, if it fails we need to see why or at least to document why we should ignore this error.

---

you may refactor this using the Finder component => https://symfony.com/doc/current/components/finder.html. In case of error, we will have a meaningful exception.